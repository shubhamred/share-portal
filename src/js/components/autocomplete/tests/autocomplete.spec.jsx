/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import AutocompleteComponent from '../index';

configure({ adapter: new Adapter() });
let AutocompleteComponentWrapper;
global.document.createRange = () => ({
  setStart: () => {},
  setEnd: () => {},
  commonAncestorContainer: {
    nodeName: 'BODY',
    ownerDocument: document
  }
});
describe('components/Autocomplete Component', () => {
  it('Renders default options', () => {
    const input = {
      onChange: jest.fn(),
      value: ''
    };
    const component = shallow(
      <AutocompleteComponent input={input} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  describe('render with Object option prop', () => {
    const props = {
      options: [{ id: 1, name: 'option1', value: 'option1' }, { id: 2, name: 'option2', value: 'option2' }],
      selector: 'value',
      selectedOption: { id: 1, name: 'option1', value: 'option1' },
      handleSelectedOption: jest.fn(),
      isArray: false,
      input: {
        onChange: jest.fn(),
        value: { id: 1, name: 'option1', value: 'option1' }
      }
    };
    // eslint-disable-next-line react/jsx-props-no-spreading
    AutocompleteComponentWrapper = mount(<AutocompleteComponent {...props} />);
    it('check props length is greater >= 2', () => {
      expect(Object.keys(AutocompleteComponentWrapper.props()).length).toBeGreaterThan(1);
    });
  });
  describe('render with single prop', () => {
    const props = {
      options: ['dummy', 'dummy2'],
      selectedOption: 'dummy',
      handleSelectedOption: jest.fn(),
      input: {
        onChange: jest.fn(),
        value: 'dummy'
      },
      isArray: true,
      required: true,
      handleInputChange: jest.fn()
    };
    AutocompleteComponentWrapper = mount(<AutocompleteComponent {...props} />);
    AutocompleteComponentWrapper.find('.autoCompleteComponent input').simulate('change', { target: { value: 'The  Redemption' } });
    AutocompleteComponentWrapper.update();
    // console.log(AutocompleteComponentWrapper.find('.autoCompleteComponent input').props());
    it('Handles the onchange handler', () => {
      expect(AutocompleteComponentWrapper.props().handleInputChange).toHaveBeenCalled();
    });
  });
});
