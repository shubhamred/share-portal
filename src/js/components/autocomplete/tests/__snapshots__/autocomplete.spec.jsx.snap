// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Autocomplete Component Renders default options 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Autocomplete))
      className="wrapper autoCompleteComponent"
      classes={
        Object {
          "inputRoot": "inputHeight",
          "root": "inputBorder",
        }
      }
      clearOnBlur={true}
      disabled={false}
      freeSolo={false}
      getOptionLabel={[Function]}
      getOptionSelected={[Function]}
      inputValue=""
      onChange={[Function]}
      onInputChange={[Function]}
      options={Array []}
      renderInput={[Function]}
      value={null}
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
